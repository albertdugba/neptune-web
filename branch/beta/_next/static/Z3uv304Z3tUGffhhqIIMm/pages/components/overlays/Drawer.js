(window.webpackJsonp=window.webpackJsonp||[]).push([[78],{"05Xt":function(e,t,n){"use strict";var r=n("cxan"),a=n("+wNj"),o=n("BFfR");n("aWzz");function i(e,t){return e.replace(new RegExp("(^|\\s)"+t+"(?:\\s|$)","g"),"$1").replace(/\s+/g," ").replace(/^\s*|\s*$/g,"")}var s=n("ERkP"),l=n.n(s),c=n("sypB"),u=function(e,t){return e&&t&&t.split(" ").forEach((function(t){return r=t,void((n=e).classList?n.classList.remove(r):"string"===typeof n.className?n.className=i(n.className,r):n.setAttribute("class",i(n.className&&n.className.baseVal||"",r)));var n,r}))},p=function(e){function t(){for(var t,n=arguments.length,r=new Array(n),a=0;a<n;a++)r[a]=arguments[a];return(t=e.call.apply(e,[this].concat(r))||this).appliedClasses={appear:{},enter:{},exit:{}},t.onEnter=function(e,n){t.removeClasses(e,"exit"),t.addClass(e,n?"appear":"enter","base"),t.props.onEnter&&t.props.onEnter(e,n)},t.onEntering=function(e,n){var r=n?"appear":"enter";t.addClass(e,r,"active"),t.props.onEntering&&t.props.onEntering(e,n)},t.onEntered=function(e,n){var r=n?"appear":"enter";t.removeClasses(e,r),t.addClass(e,r,"done"),t.props.onEntered&&t.props.onEntered(e,n)},t.onExit=function(e){t.removeClasses(e,"appear"),t.removeClasses(e,"enter"),t.addClass(e,"exit","base"),t.props.onExit&&t.props.onExit(e)},t.onExiting=function(e){t.addClass(e,"exit","active"),t.props.onExiting&&t.props.onExiting(e)},t.onExited=function(e){t.removeClasses(e,"exit"),t.addClass(e,"exit","done"),t.props.onExited&&t.props.onExited(e)},t.getClassNames=function(e){var n=t.props.classNames,r="string"===typeof n,a=r?""+(r&&n?n+"-":"")+e:n[e];return{baseClassName:a,activeClassName:r?a+"-active":n[e+"Active"],doneClassName:r?a+"-done":n[e+"Done"]}},t}Object(o.a)(t,e);var n=t.prototype;return n.addClass=function(e,t,n){var r=this.getClassNames(t)[n+"ClassName"];"appear"===t&&"done"===n&&(r+=" "+this.getClassNames("enter").doneClassName),"active"===n&&e&&e.scrollTop,this.appliedClasses[t][n]=r,function(e,t){e&&t&&t.split(" ").forEach((function(t){return r=t,void((n=e).classList?n.classList.add(r):function(e,t){return e.classList?!!t&&e.classList.contains(t):-1!==(" "+(e.className.baseVal||e.className)+" ").indexOf(" "+t+" ")}(n,r)||("string"===typeof n.className?n.className=n.className+" "+r:n.setAttribute("class",(n.className&&n.className.baseVal||"")+" "+r)));var n,r}))}(e,r)},n.removeClasses=function(e,t){var n=this.appliedClasses[t],r=n.base,a=n.active,o=n.done;this.appliedClasses[t]={},r&&u(e,r),a&&u(e,a),o&&u(e,o)},n.render=function(){var e=this.props,t=(e.classNames,Object(a.a)(e,["classNames"]));return l.a.createElement(c.a,Object(r.a)({},t,{onEnter:this.onEnter,onEntered:this.onEntered,onEntering:this.onEntering,onExit:this.onExit,onExiting:this.onExiting,onExited:this.onExited}))},t}(l.a.Component);p.defaultProps={classNames:""},p.propTypes={};t.a=p},"1Y3F":function(e,t,n){(window.__NEXT_P=window.__NEXT_P||[]).push(["/components/overlays/Drawer",function(){return n("eJ68")}])},"2AZx":function(e,t){e.exports=function(){throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.")}},"6gor":function(e,t,n){"use strict";t.a={DOWN:40,UP:38,LEFT:37,RIGHT:39,ENTER:13,ESCAPE:27,TAB:9,SPACE:32,BACKSPACE:8}},"7KeV":function(e,t,n){},"7yHk":function(e,t,n){},"91IA":function(e,t,n){},"99fE":function(e,t,n){"use strict";var r=n("XsB/"),a=n.n(r),o=n("ERkP"),i=n.n(o),s=n("7nmT");t.a=function(e){return function(t){var n=Object(o.useState)(!1),r=a()(n,2),l=r[0],c=r[1];return Object(o.useEffect)((function(){c(!0)}),[c]),l?Object(s.createPortal)(i.a.createElement(e,t),document.body):null}}},BFfR:function(e,t,n){"use strict";function r(e,t){e.prototype=Object.create(t.prototype),e.prototype.constructor=e,e.__proto__=t}n.d(t,"a",(function(){return r}))},CQzt:function(e,t,n){var r=n("IMdU");e.exports=function(e,t){if(null==e)return{};var n,a,o=r(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(o[n]=e[n])}return o}},FYE5:function(e,t){e.exports=function(e,t){(null==t||t>e.length)&&(t=e.length);for(var n=0,r=new Array(t);n<t;n++)r[n]=e[n];return r}},IMdU:function(e,t){e.exports=function(e,t){if(null==e)return{};var n,r,a={},o=Object.keys(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||(a[n]=e[n]);return a}},O94r:function(e,t,n){var r;!function(){"use strict";var n={}.hasOwnProperty;function a(){for(var e=[],t=0;t<arguments.length;t++){var r=arguments[t];if(r){var o=typeof r;if("string"===o||"number"===o)e.push(r);else if(Array.isArray(r)&&r.length){var i=a.apply(null,r);i&&e.push(i)}else if("object"===o)for(var s in r)n.call(r,s)&&r[s]&&e.push(s)}}return e.join(" ")}e.exports?(a.default=a,e.exports=a):void 0===(r=function(){return a}.apply(t,[]))||(e.exports=r)}()},P8nL:function(e,t,n){"use strict";var r=n("ERkP"),a=n.n(r),o=n("O94r"),i=n.n(o),s=n("05Xt"),l=n("99fE"),c=(n("7yHk"),n("6gor")),u=function(e){var t=e.open,n=e.children,o=e.onClose,l=e.fadeContentOnExit,u=e.fadeContentOnEnter;Object(r.useEffect)((function(){return function(){return f()}}),[]);var p=function(e){e&&(e.keyCode===c.a.ESCAPE||"Escape"===e.key)&&d(e)},d=function(e){f(),o&&o(e)},f=function(){document.removeEventListener("keydown",p)};return a.a.createElement(s.a,{in:t,appear:!0,timeout:{enter:0,exit:350},onEnter:function(){document.addEventListener("keydown",p)},onExited:d,classNames:{enter:i()({"dimmer--enter-fade":u}),enterDone:i()("dimmer--enter-done",{"dimmer--enter-fade":u}),exit:i()("dimmer--exit",{"dimmer--exit-fade":l})},unmountOnExit:!0},a.a.createElement("div",{role:"presentation",className:"dimmer",onClick:function(e){e.target===e.currentTarget&&d(e)}},n))};u.defaultProps={open:!1,children:null,onClose:null,fadeContentOnExit:!1,fadeContentOnEnter:!1},t.a=Object(l.a)(u)},SP0Y:function(e,t,n){"use strict";var r=n("eS6V"),a=n.n(r),o=n("CQzt"),i=n.n(o),s=n("ERkP"),l=n.n(s),c=n("O94r"),u=n.n(c),p=(n("j+zR"),n("91IA"),{PRIMARY:"primary",PAY:"pay",SECONDARY:"secondary",DANGER:"danger",LINK:"link"}),d=n("VehP"),f=function(e){var t=e.className,n=e.block,r=e.children,o=e.disabled,s=e.htmlType,c=e.loading,f=e.size,m=e.type,E=i()(e,["className","block","children","disabled","htmlType","loading","size","type"]),h=u()("btn btn-".concat(f),"tw-btn tw-btn-".concat(f),{"btn-loading":c,"btn-primary":m===p.PRIMARY,"btn-success":m===p.PAY,"btn-default":m===p.SECONDARY,"btn-danger":m===p.DANGER,"btn-link":m===p.LINK,"btn-block tw-btn-block":n},t);return d.a.EXTRA_SMALL,l.a.createElement("button",a()({type:s,className:h,disabled:o||c},E),r,c&&l.a.createElement("span",{className:u()("btn-loader",{"m-l-2":!n})}))};f.Type=p,f.Size={EXTRA_SMALL:d.a.EXTRA_SMALL,SMALL:d.a.SMALL,MEDIUM:d.a.MEDIUM,LARGE:d.a.LARGE},f.defaultProps={className:null,size:f.Size.MEDIUM,type:f.Type.PRIMARY,disabled:!1,block:!1,loading:!1,htmlType:"button"};t.a=f},Ttzw:function(e,t,n){var r=n("FYE5");e.exports=function(e,t){if(e){if("string"===typeof e)return r(e,t);var n=Object.prototype.toString.call(e).slice(8,-1);return"Object"===n&&e.constructor&&(n=e.constructor.name),"Map"===n||"Set"===n?Array.from(n):"Arguments"===n||/^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)?r(e,t):void 0}}},V8uO:function(e,t){e.exports=function(e){if(Array.isArray(e))return e}},VehP:function(e,t,n){"use strict";n.d(t,"a",(function(){return r}));var r={EXTRA_SMALL:"xs",SMALL:"sm",MEDIUM:"md",LARGE:"lg",EXTRA_LARGE:"xl"}},"XsB/":function(e,t,n){var r=n("V8uO"),a=n("jPt+"),o=n("Ttzw"),i=n("2AZx");e.exports=function(e,t){return r(e)||a(e,t)||o(e,t)||i()}},bRoR:function(e,t,n){"use strict";n.d(t,"a",(function(){return r})),n.d(t,"b",(function(){return a}));var r=function(){document.body.classList.add("no-scroll")},a=function(){document.body.classList.remove("no-scroll")}},eJ68:function(e,t,n){"use strict";n.r(t),n.d(t,"meta",(function(){return y})),n.d(t,"default",(function(){return N}));var r=n("cxan"),a=n("HbGN"),o=n("ERkP"),i=n.n(o),s=n("ZVZ0"),l=n("Qi1R"),c=n("O94r"),u=n.n(c),p=n("/Vl7"),d=n("sob3"),f=n("P8nL"),m=n("bRoR"),E=(n("7KeV"),n("6gor")),h=function(e){var t=e.open,n=e.position,r=e.onClose,a=e.children,s=e.headerTitle,l=e.footerContent;return Object(o.useEffect)((function(){return function(){t&&Object(m.b)()}}),[]),i.a.createElement(f.a,{open:t,onClose:r},i.a.createElement(d.a,{open:t,position:n,onEnter:function(){Object(m.a)()},onExit:function(){Object(m.b)()}},i.a.createElement("div",{className:"drawer"},i.a.createElement("div",{className:u()("drawer-header","drawer-p-x",{"drawer-header--withborder":s})},i.a.createElement("div",{role:"button",className:"drawer-header close",tabIndex:0,onClick:r,onKeyDown:function(e){(e.keyCode===E.a.ESCAPE||"Escape"===e.key||e.keyCode===E.a.ENTER||e.key===E.a.ENTER)&&r(e)}},i.a.createElement(p.h,{size:24})),s&&i.a.createElement("div",{className:"align-heading m-l-2"},i.a.createElement("h3",{className:"drawer-header--title"},s))),a&&i.a.createElement("div",{className:u()("drawer-content","drawer-p-x","drawer-p-y")},a),l&&i.a.createElement("div",{className:u()("drawer-footer","drawer-p-x")},l))))};h.defaultProps={open:!1,children:null,position:"right",onClose:null,headerTitle:null,footerContent:null};var b=h,x=n("SP0Y"),v='() => {\n  const [open, setOpen] = React.useState(false);\n  return (\n    <>\n      <Button\n        onClick={() => setDimmerState(true)}\n        disabled={false}\n        block={false}\n        onClick={() => setOpen(true)}\n      >\n        Open drawer\n      </Button>\n      <Drawer\n        open={open}\n        position="right"\n        onClose={() => setOpen(false)}\n        footerContent={\n          <Button onClick={() => setOpen(false)} block>\n            Action\n          </Button>\n        }\n        headerTitle="A title"\n      >\n        <input type="text" className="form-control" />\n        <p className="m-t-3">Cat ipsum dolor sit amet, purr when being pet.</p>\n      </Drawer>\n    </>\n  );\n};\n',y=(i.a.createElement,{name:"Drawer"}),C={meta:y},g="wrapper";function N(e){var t=e.components,n=Object(a.a)(e,["components"]);return Object(s.a)(g,Object(r.a)({},C,n,{components:t,mdxType:"MDXLayout"}),Object(s.a)("p",null,Object(s.a)("strong",{parentName:"p"},"Note:")," To use this component you must include ",Object(s.a)("inlineCode",{parentName:"p"},"neptune.css")," or ",Object(s.a)("inlineCode",{parentName:"p"},"neptune-addons.css")," in your application."),Object(s.a)(l.b,{code:v,scope:{Drawer:b,Button:x.a},mdxType:"LiveEditorBlock"}),Object(s.a)(l.a,{componentName:"Drawer",mdxType:"GeneratePropsTable"}))}N.isMDXComponent=!0},eS6V:function(e,t){function n(){return e.exports=n=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var n=arguments[t];for(var r in n)Object.prototype.hasOwnProperty.call(n,r)&&(e[r]=n[r])}return e},n.apply(this,arguments)}e.exports=n},"j+zR":function(e,t){e.exports=function(e,t){return function(n,r,a){if("function"!==typeof e)return new Error("Invalid react-required-if prop type supplied to "+a+". Validation failed.");if("function"!==typeof t)return new Error("Invalid react-required-if condition supplied to "+a+". Validation failed.");var o=t(n)?e.isRequired:e;return o.apply(this,arguments)}}},"jPt+":function(e,t){e.exports=function(e,t){if("undefined"!==typeof Symbol&&Symbol.iterator in Object(e)){var n=[],r=!0,a=!1,o=void 0;try{for(var i,s=e[Symbol.iterator]();!(r=(i=s.next()).done)&&(n.push(i.value),!t||n.length!==t);r=!0);}catch(l){a=!0,o=l}finally{try{r||null==s.return||s.return()}finally{if(a)throw o}}return n}}},sob3:function(e,t,n){"use strict";var r=n("eS6V"),a=n.n(r),o=n("CQzt"),i=n.n(o),s=n("ERkP"),l=n.n(s),c=n("O94r"),u=n.n(c),p=n("05Xt"),d=(n("yiBs"),function(e){var t=e.open,n=e.position,r=e.children,o=e.slidingPanelPositionFixed,s=e.showSlidingPanelBorder,c=i()(e,["open","position","children","slidingPanelPositionFixed","showSlidingPanelBorder"]);return l.a.createElement(p.a,a()({},c,{in:t,timeout:{enter:0,exit:350},classNames:u()("sliding-panel--open-".concat(n),s&&"sliding-panel--border-".concat(n),{"sliding-panel--fixed":o},"sliding-panel"),appear:!0,unmountOnExit:!0}),l.a.createElement("div",{className:"sliding-panel"},r))});d.defaultProps={open:!1,slidingPanelPositionFixed:!1,showSlidingPanelBorder:!1,position:"left",children:null},t.a=d},sypB:function(e,t,n){"use strict";var r=n("+wNj"),a=n("BFfR"),o=(n("aWzz"),n("ERkP")),i=n.n(o),s=n("7nmT"),l=n.n(s),c=!1,u=i.a.createContext(null),p="unmounted",d="exited",f="entering",m="entered",E=function(e){function t(t,n){var r;r=e.call(this,t,n)||this;var a,o=n&&!n.isMounting?t.enter:t.appear;return r.appearStatus=null,t.in?o?(a=d,r.appearStatus=f):a=m:a=t.unmountOnExit||t.mountOnEnter?p:d,r.state={status:a},r.nextCallback=null,r}Object(a.a)(t,e),t.getDerivedStateFromProps=function(e,t){return e.in&&t.status===p?{status:d}:null};var n=t.prototype;return n.componentDidMount=function(){this.updateStatus(!0,this.appearStatus)},n.componentDidUpdate=function(e){var t=null;if(e!==this.props){var n=this.state.status;this.props.in?n!==f&&n!==m&&(t=f):n!==f&&n!==m||(t="exiting")}this.updateStatus(!1,t)},n.componentWillUnmount=function(){this.cancelNextCallback()},n.getTimeouts=function(){var e,t,n,r=this.props.timeout;return e=t=n=r,null!=r&&"number"!==typeof r&&(e=r.exit,t=r.enter,n=void 0!==r.appear?r.appear:t),{exit:e,enter:t,appear:n}},n.updateStatus=function(e,t){if(void 0===e&&(e=!1),null!==t){this.cancelNextCallback();var n=l.a.findDOMNode(this);t===f?this.performEnter(n,e):this.performExit(n)}else this.props.unmountOnExit&&this.state.status===d&&this.setState({status:p})},n.performEnter=function(e,t){var n=this,r=this.props.enter,a=this.context?this.context.isMounting:t,o=this.getTimeouts(),i=a?o.appear:o.enter;!t&&!r||c?this.safeSetState({status:m},(function(){n.props.onEntered(e)})):(this.props.onEnter(e,a),this.safeSetState({status:f},(function(){n.props.onEntering(e,a),n.onTransitionEnd(e,i,(function(){n.safeSetState({status:m},(function(){n.props.onEntered(e,a)}))}))})))},n.performExit=function(e){var t=this,n=this.props.exit,r=this.getTimeouts();n&&!c?(this.props.onExit(e),this.safeSetState({status:"exiting"},(function(){t.props.onExiting(e),t.onTransitionEnd(e,r.exit,(function(){t.safeSetState({status:d},(function(){t.props.onExited(e)}))}))}))):this.safeSetState({status:d},(function(){t.props.onExited(e)}))},n.cancelNextCallback=function(){null!==this.nextCallback&&(this.nextCallback.cancel(),this.nextCallback=null)},n.safeSetState=function(e,t){t=this.setNextCallback(t),this.setState(e,t)},n.setNextCallback=function(e){var t=this,n=!0;return this.nextCallback=function(r){n&&(n=!1,t.nextCallback=null,e(r))},this.nextCallback.cancel=function(){n=!1},this.nextCallback},n.onTransitionEnd=function(e,t,n){this.setNextCallback(n);var r=null==t&&!this.props.addEndListener;e&&!r?(this.props.addEndListener&&this.props.addEndListener(e,this.nextCallback),null!=t&&setTimeout(this.nextCallback,t)):setTimeout(this.nextCallback,0)},n.render=function(){var e=this.state.status;if(e===p)return null;var t=this.props,n=t.children,a=Object(r.a)(t,["children"]);if(delete a.in,delete a.mountOnEnter,delete a.unmountOnExit,delete a.appear,delete a.enter,delete a.exit,delete a.timeout,delete a.addEndListener,delete a.onEnter,delete a.onEntering,delete a.onEntered,delete a.onExit,delete a.onExiting,delete a.onExited,"function"===typeof n)return i.a.createElement(u.Provider,{value:null},n(e,a));var o=i.a.Children.only(n);return i.a.createElement(u.Provider,{value:null},i.a.cloneElement(o,a))},t}(i.a.Component);function h(){}E.contextType=u,E.propTypes={},E.defaultProps={in:!1,mountOnEnter:!1,unmountOnExit:!1,appear:!1,enter:!0,exit:!0,onEnter:h,onEntering:h,onEntered:h,onExit:h,onExiting:h,onExited:h},E.UNMOUNTED=0,E.EXITED=1,E.ENTERING=2,E.ENTERED=3,E.EXITING=4;t.a=E},yiBs:function(e,t,n){}},[["1Y3F",0,1,2,3,7,4,5]]]);