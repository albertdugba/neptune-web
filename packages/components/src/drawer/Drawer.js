import classNames from 'classnames';
import PropTypes from 'prop-types';

import CloseButton from '../common/closeButton';
import { Position } from '../common/propsValues/position';
import Dimmer from '../dimmer';
import SlidingPanel from '../slidingPanel';
import { logActionRequiredIf } from '../utilities';

const Drawer = ({ children, footerContent, headerTitle, onClose, open, position }) => {
  logActionRequiredIf(
    'Drawer now expects `onClose`, and will soon make this prop required. Please update your usage to provide it.',
    !onClose,
  );

  return (
    <Dimmer open={open} onClose={onClose}>
      <SlidingPanel open={open} position={position}>
        <div className="np-drawer">
          <div
            className={classNames('np-drawer-header', 'np-drawer-p-x', {
              'np-drawer-header--withborder': headerTitle,
            })}
          >
            <CloseButton className="np-drawer-header" onClick={onClose} />
            {headerTitle && (
              <div className="align-heading m-l-2">
                <div className="np-drawer-header--title h3">{headerTitle}</div>
              </div>
            )}
          </div>
          {children && (
            <div className={classNames('np-drawer-content', 'np-drawer-p-x', 'np-drawer-p-y')}>
              {children}
            </div>
          )}
          {footerContent && (
            <div className={classNames('np-drawer-footer', 'np-drawer-p-x')}>{footerContent}</div>
          )}
        </div>
      </SlidingPanel>
    </Dimmer>
  );
};

Drawer.propTypes = {
  /** The content to appear in the drawer body. */
  children: PropTypes.node,
  /** The content to appear in the drawer footer. */
  footerContent: PropTypes.node,
  /** The content to appear in the drawer header. */
  headerTitle: PropTypes.string,
  /** The action to perform on close click. */
  onClose: PropTypes.func,
  /** The status of Drawer either open or not. */
  open: PropTypes.bool,
  /** The placement of Drawer on the screen either left or right. */
  position: PropTypes.oneOf(['left', 'right']),
};

Drawer.defaultProps = {
  children: null,
  footerContent: null,
  headerTitle: null,
  onClose: null,
  open: false,
  position: Position.RIGHT,
};

export default Drawer;
