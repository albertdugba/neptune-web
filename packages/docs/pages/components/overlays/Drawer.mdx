import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
import { Drawer, Button, Position } from '@transferwise/components';
import code from '../../../liveEditorCode/drawer.code';

**Note:** To use this component you must include `neptune.css` or `neptune-addons.css` in your application.

<LiveEditorBlock code={code} scope={{ Drawer, Button, Position }} />
<GeneratePropsTable componentName="Drawer" />

export const meta = {
  name: 'Drawer',
};
