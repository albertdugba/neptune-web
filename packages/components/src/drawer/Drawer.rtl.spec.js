import { render, screen, userEvent } from '../test-utils';

import Drawer from './Drawer';

jest.mock(
  '../dimmer',
  () =>
    function ({ open, children }) {
      return open ? <div className="dimmer">{children}</div> : null;
    },
);

jest.mock(
  '../slidingPanel',
  () =>
    function ({ open, children }) {
      return open ? <div className="sliding-panel">{children}</div> : null;
    },
);

describe('Drawer', () => {
  const props = {
    onClose: jest.fn(),
    open: true,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders content when open', () => {
    const { container } = render(<Drawer {...props}>content</Drawer>);

    expect(container).toMatchSnapshot();
  });

  it("doesn't render content when closed", () => {
    const { container } = render(
      <Drawer {...props} open={false}>
        content
      </Drawer>,
    );

    expect(container).toMatchSnapshot();
  });

  it('calls onClose when user clicks close button', () => {
    render(<Drawer {...props}>content</Drawer>);
    expect(props.onClose).not.toHaveBeenCalled();
    userEvent.click(getCloseButton());
    expect(props.onClose).toHaveBeenCalledTimes(1);
  });

  const getCloseButton = () => screen.getByLabelText('Close');
});
