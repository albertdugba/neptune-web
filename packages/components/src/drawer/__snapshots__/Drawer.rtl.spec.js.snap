// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer doesn't render content when closed 1`] = `<div />`;

exports[`Drawer renders content when open 1`] = `
<div>
  <div
    class="dimmer"
  >
    <div
      class="sliding-panel"
    >
      <div
        class="np-drawer"
      >
        <div
          class="np-drawer-header"
        >
          <button
            aria-label="Close"
            class="np-close-button close btn-link text-no-decoration np-close-button--large"
            type="button"
          >
            <span
              aria-hidden="true"
              class="tw-icon tw-icon-cross "
              role="presentation"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="24"
                width="24"
              >
                <path
                  d="M12 13.414l7.293 7.293 1.414-1.414L13.414 12l7.293-7.293-1.414-1.414L12 10.586 4.707 3.293 3.293 4.707 10.586 12l-7.293 7.293 1.414 1.414L12 13.414z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="np-drawer-content"
        >
          content
        </div>
      </div>
    </div>
  </div>
</div>
`;
